spec/requests/providers_spec.rb

```diff
@@ -1536,19 +1548,30 @@ def gen_import_request
     context 'provider_class=provider' do
       let(:generic_provider) { FactoryGirl.create(:provider) }
       let(:attr) { FactoryGirl.create(:custom_attribute) }
-      let(:url) do
-        api_provider_custom_attributes_url(nil, generic_provider) + '?provider_class=provider'
-      end
 
       it 'cannot add a custom_attribute' do
         api_basic_authorize subcollection_action_identifier(:providers, :custom_attributes, :add, :post)
-        post(url, :params => gen_request(:add, :name => 'x'))
+
+        post(
+          api_provider_custom_attributes_url(nil, generic_provider),
+          :headers => {"QUERY_STRING" => "provider_class=provider"},
+          :params  => {:action => "add", :resource => {:name => "x"}}
+        )
+
         expect_bad_request("#{generic_provider.class.name} does not support management of custom attributes")
       end
 
       it 'cannot edit custom_attribute' do
         api_basic_authorize subcollection_action_identifier(:providers, :custom_attributes, :edit, :post)
-        post(url, :params => gen_request(:edit, :href => api_provider_custom_attribute_url(nil, generic_provider, attr)))
+        post(
+          api_provider_custom_attributes_url(nil, generic_provider),
+          :headers => {"QUERY_STRING" => "provider_class=provider"},
+          :params  => {
```

Comment:
would prefer to keep using gen_request in all the above :params as to not deviate with the rest of the specs.  Thanks.

Reply:
Personally, I think `gen_request` should be slowly deprecated in favor of a more transparent request declaration as the above, as we have started to do elsewhere: 1) `gen_request` isn't really descriptive, 2) it obscures important details about how the request is structured. A test, as well as exercising code in the system, should also serve as a clear example, and these details are crucial to understanding what was sent in the request. As such I'd much rather deal with inconsistency while seizing opportunities such as this to slowly move toward a more clear example as we have to change the surrounding code anyway.